python/mxnet/context.py

```diff
@@ -109,15 +112,37 @@ def __repr__(self):
         return self.__str__()
 
     def __enter__(self):
-        self._old_ctx = Context.default_ctx
-        Context.default_ctx = self
+        if not hasattr(Context._default_ctx, "value"):
```

Comment:
why do you need this? Isn't it set with https://github.com/apache/incubator-mxnet/pull/10833/files#diff-1a5e06031378f44204fd1da1fffc0b07R145

Reply:
what about a situation like below:
```
import threading
import mxnet as mx

def test_context():
    ctx_list = []
    def f():
        mx.context.Context.default_ctx
        ctx_list.append(Context.default_ctx)
    thread = threading.Thread(target=f)
    thread.start()
    thread.join()

if __name__ == '__main__':
    test_context()
```

Without hasattr this doesn't work.